packages/ibm-products/src/components/Datagrid/Datagrid/addons/Filtering/OverflowCheckboxes.js

```diff
@@ -0,0 +1,78 @@
+import { Button, Checkbox, FormGroup } from '@carbon/react';
+import React, { useState } from 'react';
+
+import PropTypes from 'prop-types';
+import { componentClass } from './FilterPanel';
+
+const OverflowCheckboxes = ({
+  components,
+  filtersState,
+  column,
+  setFiltersState,
+  type,
+  applyFilters,
+}) => {
+  const [showAllItems, setShowAllItems] = useState(false);
+  const firstFiveItems = filtersState[column].value.slice(0, 5);
+  const restOfTheItems = filtersState[column].value.slice(
+    5,
+    filtersState[column].value.length - 1
+  );
+
+  const renderCheckbox = (option) => (
+    <Checkbox
+      key={option.labelText}
+      {...option}
+      onChange={(isSelected) => {
+        const checkboxCopy = filtersState[column].value;
+        const foundCheckbox = checkboxCopy.find(
+          (checkbox) => checkbox.value === option.value
+        );
+        foundCheckbox.selected = isSelected;
+        setFiltersState({
+          ...filtersState,
+          [column]: {
+            value: checkboxCopy,
+            type,
+          },
+        });
+        applyFilters({
+          column,
+          value: [...filtersState[column].value],
+          type,
+        });
+        option.onChange?.(isSelected);
+      }}
+      checked={option.selected}
+      disabled={option.disabled}
+    />
+  );
+
+  return (
+    <FormGroup {...components.FormGroup}>
+      {firstFiveItems.map(renderCheckbox)}
+      {showAllItems && restOfTheItems.map(renderCheckbox)}
+      <Button
+        className={`${componentClass}__view-all-button`}
+        kind="ghost"
+        size="sm"
+        onClick={() => setShowAllItems(!showAllItems)}
+      >
+        {showAllItems
+          ? 'View less'
```

Comment:
These are going to need to be passed in so we can support translation, I would give them a default value though so it defaults to `View less` and `View all`.

Reply:
Fixed 🔧